File: python/sparseArrays.py
def matchingStrings(strings, queries):
    # Write your code here
    def str_count(arr, text):
      count = 0
      for x in arr:
        if x == text:
          count+= 1
      return count
    arr_of_counts = map(lambda query: str_count(strings, query), queries)
    return list(arr_of_counts)

result = matchingStrings(["ab", "ab", "abc"], ["ab", "abc", "bc"])
print(result)

File: javascript/sparseArrays.js
function matchingStrings(strings, queries) {
  // Write your code here
  const str_count = (arr, str) => {
    let count = 0;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i] == str) {
        count += 1;
      }
    }
    return count;
  };
  const arr_of_counts = queries.map((query) => str_count(strings, query));
  return arr_of_counts;
}

const result = matchingStrings(["ab", "ab", "abc"], ["ab", "abc", "bc"]);
console.log(result);
